flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/ExecutionGraph.java

```diff
@@ -101,6 +102,8 @@ void enableCheckpointing(
 
     void setJsonPlan(String jsonPlan);
 
+    void setJsonStreamGraph(JsonStreamGraph jsonStreamGraph);
```

Comment:
I prefer to retrieve the JSON of the stream graph from `ExecutionPlanSchedulingContext`. For example, we could add a method `getJsonStreamGraph` in `ExecutionPlanSchedulingContext`. This context will return the JSON held by `AdaptiveGraphManager`. When the stream graph is updated by the `DefaultStreamGraphContext`, the AdaptiveGraphManager should update this JSON.

With this update, we could avoid manually passing this JSON to the execution graph and keep this change simple.

Reply:
Great tips! `ExecutionPlanSchedulingContext` really helps a lot. I have remove the unexpected `setJsonStreamGraph` methods from `ExecutionGraph` in the revised version.